src/server.ts

```diff
@@ -58,9 +59,11 @@ export const createServer = async (config: ServerConfig, routing: Routing) => {
         ...derivedConfig,
         abortOnLimit: false,
         parseNested: true,
-        limitHandler: limitError && (({}, {}, next) => next(limitError)),
       }),
     );
+    if (limitError) {
+      app.use(createUploadFailueHandler(limitError));
```

Comment:
this approach can handle only `fileSize` of `limits`.

Reply:
but that turned out to be OK because other limits do not trigger `limitHandler` —I checked it